test/token/ERC721/extensions/ERC721Consecutive.test.js

```diff
@@ -0,0 +1,128 @@
+const { constants, expectEvent, expectRevert } = require('@openzeppelin/test-helpers');
+const { expect } = require('chai');
+
+const ERC721ConsecutiveMock = artifacts.require('ERC721ConsecutiveMock');
+
+contract('ERC721Consecutive', function (accounts) {
+  const [ user1, user2, user3, receiver ] = accounts;
+
+  const name = 'Non Fungible Token';
+  const symbol = 'NFT';
+  const batches = [
+    { receiver: user1, amount: 0 },
+    { receiver: user1, amount: 3 },
+    { receiver: user2, amount: 5 },
+    { receiver: user3, amount: 0 },
+    { receiver: user1, amount: 7 },
+  ];
+
+  beforeEach(async function () {
+    this.token = await ERC721ConsecutiveMock.new(
+      name,
+      symbol,
+      batches.map(({ receiver }) => receiver),
+      batches.map(({ amount }) => amount),
+    );
+  });
+
+  describe('minting during construction', function () {
+    it('events are emitted at construction', async function () {
+      let first = 0;
+
+      for (const batch of batches) {
+        if (batch.amount == 0) continue;
+
+        await expectEvent.inTransaction(this.token.transactionHash, this.token, 'ConsecutiveTransfer', {
+          fromTokenId: web3.utils.toBN(first),
+          toTokenId: web3.utils.toBN(first + batch.amount - 1),
+          fromAddress: constants.ZERO_ADDRESS,
+          toAddress: batch.receiver,
+        });
+
+        first += batch.amount;
+      }
+    });
+
+    it('ownership is set', async function () {
+      const owners = batches.flatMap(({ receiver, amount }) => Array(amount).fill(receiver));
+
+      for (const tokenId in owners) {
+        expect(await this.token.ownerOf(tokenId))
+          .to.be.equal(owners[tokenId]);
+      }
+    });
+
+    it('balance & voting poser are set', async function () {
+      for (const account of accounts) {
+        const balance = batches
+          .filter(({ receiver }) => receiver === account)
+          .map(({ amount }) => amount)
+          .reduce((a, b) => a + b, 0);
+
+        expect(await this.token.balanceOf(account))
+          .to.be.bignumber.equal(web3.utils.toBN(balance));
+
+        expect(await this.token.getVotes(account))
```

Comment:
I think it would be better and more explicit if we removed self-delegation from the mock and did it as part of the test, after testing that initially voting power is zero.

Reply:
We should do both. I'll keep self delegation in the constructor, but only for a subset of the recipient. The other will delegate later.

Self delegation in the constructor is the only way I found to test correct delegation during mint (because the mint is only during construction)